api/src/main/java/com/datatorrent/api/LocalMode.java

```diff
@@ -18,34 +18,64 @@
  */
 package com.datatorrent.api;
 
-import java.util.Iterator;
-import java.util.ServiceLoader;
-
+import org.apache.apex.api.Launcher;
 import org.apache.hadoop.conf.Configuration;
 
+import com.google.common.base.Throwables;
+
 /**
  * Local mode execution for single application
  *
+ * @deprecated
  * @since 0.3.2
  */
-public abstract class LocalMode
+@Deprecated
+public abstract class LocalMode<H extends Launcher.AppHandle> extends Launcher<H>
 {
 
   /**
+   * Parameter to specify the time after which the application will be shutdown; pass 0 to run indefinitely.
+   */
+  public static final Attribute<Long> RUN_MILLIS = new Attribute<Long>(0L);
+
+  /**
+   * Parameter to launch application asynchronously and return from launch immediately.
+   */
+  public static final Attribute<Boolean> RUN_ASYNC = new Attribute<Boolean>(false);
+
+  /**
+   * Parameter to enable or disable heartbeat monitoring.
+   */
+  public static final Attribute<Boolean> HEARTBEAT_MONITORING = new Attribute<Boolean>(true);
+
+  /**
+   * Parameter to serialize DAG before launch.
+   */
+  public static final Attribute<Boolean> SERIALIZE_DAG = new Attribute<Boolean>(false);
```

Comment:
OK, we can leave it as it is and deal with it as follow-up. If the user wants to check and other things, then providing access to the DAG seems sufficient.